### ISSUE TITLE
Does not reject multiple `--image` parameters

### ISSUE THREAD

**Issue by jchesterpivotal** [knative/client#533]
## What version of Knative Client?

> Paste output of 'kn version'

```
$ kn version
Version:      v0.10.0
Build Date:   2019-11-07 15:02:21
Git Revision: 22c65d6
Supported APIs:
- serving.knative.dev/v1alpha1 (knative-serving v0.10.0)
```

## What version of Knative Serving running on your cluster?

0.10

## Expected Behavior

When I type `kn update service --image example.com/first --image example.com/second`, I expect to be told that I cannot provide two images.

## Actual Behavior

`kn` accepts the command line without error or warning. It applies the last-provided argument. In the example above this means that the updated service has `example.com/second` as its image.

## Steps to Reproduce the Problem

1. Create a service.
1. Update the service, using two different `--image` parameters.
1. Observe that no error occurs.
1. Use describe to observe that second parameter is applied.

**Comment by rhuss** [comment]
Indeed, that's a bug. Thanks for reporting.

**Comment by itsmurugappan** [comment]
same issue with the all the flags. I could give like 
1. --requests-cpu 50m --requestspu 60m
2. --port=8080 --port=8081

**Comment by itsmurugappan** [comment]
@rhuss  I tried couple of other popular CLI's they dont seem to have a catch for this. For example , the below works, creates deployment with the first image
```
kubectl create deployment test --image=image1 --image=image2
```

**Comment by rhuss** [comment]
@itsmurugappan thanks for the investigations. I guess this is kind of a limitation of Cobra. I wonder though whether there is a workaround there for validating that only a single argument can be used. Will look into this later this week as I have to get more knowledge about Cobra and whether it is still suitable when tackling #496 which requires support for dynamically generated options.

**Comment by duglin** [comment]
I believe Cobra supports flags that can be specified multiple times, right? Like `--env`. If so, we could make `--image` one of those but then check in to code to make sure exactly one value was provided.

**Comment by dsimansk** [comment]
@rhuss the original issue with `--image` flag is addressed in #647. However, per @itsmurugappan's review and suggestion there other flags that should be treated as unique. 
Should I create a new issue to collect the list at least the most obvious/important ones?

Probably it would be worth to contribute flag feature e.g. `UniqueStringVar` back to `spf13/pflag` library, wdyt?

**Comment by rhuss** [comment]
I think its a good idea to also check our other flags.

wrt/ to contributing back: In principle that would be the way to go, but it looks like that `spf13/cobra` (and probably also "flags") is totally dead as a project. We are waiting for a PR to be merged over 6 months now, without any feedback yet: https://github.com/spf13/cobra/pull/899

So I don't think its worth the effort.
